numba/np/arraymath.py

```diff
@@ -665,6 +665,128 @@ def _trivial_isnan(x):
         return _trivial_isnan
 
 
+@overload(np.iscomplex)
+def np_iscomplex(x):
+    if type_can_asarray(x):
+        def impl(x):
+            ax = np.asarray(x) # NumPy uses asanyarray here!
+            return ax.imag != 0
+    else:
+        def impl(x):
+            return False
+    return impl
+
+
+@overload(np.isreal)
+def np_isreal(x):
+    if type_can_asarray(x):
+        def impl(x):
+            ax = np.iscomplex(x)
+            return np.invert(ax)
+    else:
+        def impl(x):
+            return False
+    return impl
+
+
+@overload(np.iscomplexobj)
+def iscomplexobj(x):
+    # Implementation based on NumPy
+    # https://github.com/numpy/numpy/blob/d9b1e32cb8ef90d6b4a47853241db2a28146a57d/numpy/lib/type_check.py#L282-L320
+    dt = determine_dtype(x)
+    iscmplx = np.issubdtype(dt, np.complexfloating)
+    return lambda x: iscmplx
+
+
+@overload(np.isrealobj)
+def isrealobj(x):
+    # Return True if x is a not complex type or an array of complex numbers.
+    # Implementation based on NumPy
+    # https://github.com/numpy/numpy/blob/ccfbcc1cd9a4035a467f2e982a565ab27de25b6b/numpy/lib/type_check.py#L290-L322
+    def impl(x):
+        return not np.iscomplexobj(x)
+    return impl
+
+
+@overload(np.isscalar)
+def np_isscalar(num):
+    if isinstance(num, (types.Number, types.UnicodeType, types.Boolean)):
+        def impl(num):
+            return True
+    else:
+        def impl(num):
+            return False
+    return impl
+
+
+@overload(np.isclose)
+def np_isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
```

Comment:
This implementation looks like the NumPy one so I think a reference with SHA in the URL would be good to add to say the implementation is based on it.